pkg/transform/snowplow_collector_tstamp.go

```diff
@@ -0,0 +1,33 @@
+package transform
+
+import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/snowplow/snowbridge/pkg/models"
+)
+
+// CollectorTstampTransformation returns a transformation function attaching collector timestamp to the input message
+// This transformation is not like other configurable transformations - it's enabled/disabled based on top-level metric configuration toggle (`metrics.enable_e2e_latency`)
+// It doesn't produce invalid data in case of errors - it logs a warning and proceeds with input data as nothing happened.
+func CollectorTstampTransformation() TransformationFunction {
+	return func(message *models.Message, interState interface{}) (*models.Message, *models.Message, *models.Message, interface{}) {
+		parsedEvent, err := IntermediateAsSpEnrichedParsed(interState, message)
+		if err != nil {
+			log.Warnf("Error while extracting 'collector_tstamp': %s", err)
```

Comment:
nit: I would make this error different from the other. In the unlikely event that we do encounter this in production, it would be helpful to know if the issue arose in parsing the event or later than that.